db/migrate/20210627154945_create_bamposts.rb

```diff
@@ -0,0 +1,10 @@
+class CreateBamposts < ActiveRecord::Migration[6.1]
+  def change
+    create_table :bamposts do |t|
+      t.text :content
+      t.integer :user_id
```

Comment:
Have you added foreign_key constraint to user_id column of your Bamposts table?  When you created Bamposts table did you use the following command?
`rails generate model Bampost user:references`
If so,  your migration file should look like this: 

```
class Bamposts < ActiveRecord::Migration[6.0]
  def change
    create_table :bamposts do |t|
      t.references :user, null: false, foreign_key: true

      t.timestamps
    end
  end
end
```

Not sure if this already works for you as in if you can relate your Users table to Bamposts table - I think it might thanks to the belongs_to and has_many associations, but the problem is when the user is removed, the posts records related to the removed user might stay in the posts table..

If you want to know more about it - here is a long explanation:

Normally you would add user_id constraint to your table either by using the above command when you create your model for the first time or by using `rails g migration AddUserToBamposts user:references` if you are adding user_id column later on  - the latter command should create the following migration file for you: 

```
class AddUserToBamposts < ActiveRecord::Migration
  def change
    add_reference :bamposts, :user, index: true
    add_foreign_key :bamposts, :users
  end
end
```

The above command basically does 3 things: 
1) adds user_id column to bamposts table 
2) adds user_id as a foreign key - so now bamposts will extract user_id info from the id column of users table, it would also not allow user_id column to be null (hence null:false)
3) adds an index on user_id column so it is easier to look for - for when we need to find a post based on user_id/when we want to find the owner of the post.

You have already added user_id column to bamposts table, but I think you have not added a foreign key constraint yet. You have also associated these tables together using: belongs_to and has_many, but we need a foreign key reference to 'relate' these tables. To add a foreign key constraint to an already existing column (user_id), you can type something like this in your command line:

`rails generate migration AddForeignKeyToBamposts`. This will create an empty migration file for you. You can then go inside the migration file and add ` add_foreign_key :bamposts, :users, on_delete: :cascade, validate: false` after `def change`, so it looks like this: 

```
class AddForeignKeyToBamposts < ActiveRecord::Migration[6.0]
  def change
    add_foreign_key :bamposts, :users, on_delete: :cascade, validate: false
  end
end
```

Don't forget to run `rails db:migrate` after creating your migrations.  Let me know if you need me to explain any of this - also you can also find a good explanation [here](https://docs.gitlab.com/ee/development/database/add_foreign_key_to_existing_column.html):